### ISSUE TITLE
don't require IP address or fully qualified domain name

### ISSUE THREAD

**Issue by jmissig** [oogje/homebridge-i6-bigAssFans#6]
**Is your feature request related to a problem? Please describe:**
My fans did not have fixed IP addresses. I modified my network set up to get my DHCP server to assign them fixed IPs for now, but wish I didn’t have to.

**Describe the solution you'd like:**
No idea what exists in this version of the Big Ass Fans protocol, but [the older Haiku fans plugin](https://github.com/sean9keenan/homebridge-bigAssFans) supported discovering fans if I told it how many to expect.

**Comment by jmissig** [comment]
Wanted to add: Thank you very much for making this plugin!

@pponce gave me a heads-up that this plugin existed before I had updated my Haiku fans. I got them updated, got them fixed IPs, and now they’re working great with this plugin.

I have a Haiku H-series, an I-series, and an I-series with a light.

**Comment by oogje** [comment]
I'm all for this.  If someone tells me what message the Big Ass Fans app sends out to trigger discovery I'll give it a shot that much quicker.

I could maybe address the fans using their "hostname" rather than IP if that would helpful in the meantime.  I just noticed I can connect to mine as BigFan.local.

PS - thanks

**Comment by jmissig** [comment]
Yep it’s in no way urgent for me now, but it’s a step that I think would help others who want to get started using these fans with Homebridge. I’ll share if I figure anything out.

**Comment by oogje** [comment]
Turns out (per my test just now) if you use the fan's hostname instead of an IP address in the config, it works just as well.

**Comment by jmissig** [comment]
Interesting! I’m not having much luck with that just yet.

```
(julian@saga)=(~)=(% host 172.16.1.238
238.1.16.172.in-addr.arpa domain name pointer Entryway.
(julian@saga)=(~)=(% ping Entryway.local
ping: Entryway.local: Name or service not known
(julian@saga)=(~)=(% ping Entryway.local.
ping: Entryway.local.: Name or service not known
(julian@saga)=(~)=(% ping Entryway       
ping: Entryway: Temporary failure in name resolution
```

**Comment by oogje** [comment]
@pponce discovered his Haiku's hostname was Haiku-xx:xx:xx where his fan name was Haiku and after the dashes the last 3 octets of his MAC address.  He found it using either Discovery or wifiman iPhone apps.

**Comment by jmissig** [comment]
Thanks. I’m looking at what my Ubuntu NUC is telling me from `avahi-browse`. I see `Fanname xx:xx:xx` advertised but haven’t been able to figure out the mapping to the hostname I can use for `ping`. It does seem like a bug (on the Haiku’s end) if the advertised reverse host for address doesn’t map directly.

**Comment by oogje** [comment]
Lol, you just stepped outside my comfort zone.

**Comment by pponce** [comment]
@jmissig have you tried to ping 
Fanname-xx:xx:xx.local
I also see it advertised as “Fanname xx:xx:xx” but the above ping works.

**Comment by jmissig** [comment]
@pponce yeah trying various combinations of that now — that seems like the intended mapping — but they’re all failing. Looking more like a bug in my mDNS setup on this computer at the moment 🤷

```
+   eno1 IPv4 Entryway XX:XX:XX                             Web Site             local
=   eno1 IPv4 Entryway XX:XX:XX                             Web Site             local
   hostname = [Entryway-XX\058XX\058XX.local]
   address = [172.16.1.238]
   port = [31416]
   txt = ["mac=XX:XX:XX:XX:XX:XX" "path=/FW000007-GEN3"]
```

Edit: For anyone coming across this thread, if you have a Linux box using avahi for mDNS, the command to generate the above is `avahi-browse -r _http._tcp`

**Comment by jmissig** [comment]
Oh! It works now. `ping Entryway-XX:XX:XX.local` is succeeding. I tried that before and it was failing. mDNS it is a mystery.

Edit: For the record, Big Ass Fans folk if you’re reading this, I still think something in your mDNS advertising is incorrect if the IP reverse lookup is yielding a different hostname here:
```
(julian@saga)=(~)=(% ping Entryway-XX:XX:XX.local     
PING Entryway-XX:XX:XX.local (172.16.1.238) 56(84) bytes of data.
64 bytes from Entryway (172.16.1.238): icmp_seq=1 ttl=128 time=4.27 ms
```

**Comment by oogje** [comment]
Cool.  Thanks for confirming this.

**Comment by oogje** [comment]
Now I have to ask, what's` port = [31416]` in the above `avahi-browse -r _http._tcp` output?  I only know about port 31415.

**Comment by jmissig** [comment]
Seems to be an HTTP session (makes sense, since it’s advertising an HTTP address in mDNS!). I’m trying some basic GET requests (at least, what I think are basic requests) and they’re all returning 400.

**Comment by oogje** [comment]
interesting 🤔

**Comment by jmissig** [comment]
Definitely! I’ve not been able to yield anything of interest yet, but there’s _something_ responding negatively to HTTP requests on that port.

**Comment by oogje** [comment]
perhaps the ghost of firmware past (or yet to come).

**Comment by pponce** [comment]
I use the iOS “Discovery” app.
Within that I go to the 
“world wide web http, _http._tcp.” row.
When I click into that, I see my fans.
And if I click on a fan, amongst other details i see
FanName-XX:Xx:XX.local:31416
so I also see this 31416 port.

**Comment by jmissig** [comment]
I’ve been successfully using the `Entryway-XX:XX:XX.local` type format since, well, April 23 2022 presumably.

Today some combination of updates seems to have made that stop working. I repeatedly get:
```
[7/22/2023, 7:15:42 AM] [homebridge-i6-bigassfans] Cocktail Room Overhead Fan (CocktailRoom-XX:XX:XX.local) : Unhandled network error: ENOTFOUND.  Attempting reconnect in 512 seconds.
```

Setup
- Ubunutu 22.04.2 LTS (didn’t change, but I perform regular apt full-upgrade)
- Nodejs v18.17.0 (updated just now)
- Homebridge v1.6.1 (updated recently)
- homebridge-i6-bigassfans v0.5.4 (updated recently)

Things I tried:
- Confirmed I can ping `CocktailRoom-XX:XX:XX.local` and all the rest from the computer running Homebridge just fine.
- Switched Homebridge between Avahi and Ciao, no change.

For now I’ve switched to resolved IP address, which works.
